src/xopen/__init__.py

```diff
@@ -336,6 +328,25 @@ def __iter__(self):
     def __next__(self) -> AnyStr:
         return self._file.__next__()
 
+    def _wait_for_output_or_process_exit(self):
+        """
+        Wait for the process to produce at least some output, or has exited.
+        """
+        # The program may crash due to a non-existing file, internal error etc.
+        # In that case we need to check. However the 'time-to-crash' differs
+        # between programs. Some crash faster than others.
+        # Therefore we peek the first character(s) of stdout. Peek will return at
+        # least one byte of data, unless the buffer is empty or at EOF. If at EOF,
+        # we should wait for the program to exit. This way we ensure the program
+        # has at least decompressed some output, or stopped before we continue.
+
+        # stdout is io.BufferedReader if set to PIPE
+        while True:
+            first_output = self.process.stdout.peek(1)  # type: ignore
+            if first_output or self.process.poll() is not None:
+                break
+            time.sleep(0.1)
```

Comment:
Can you make this `0.01`? When opening lots of files 0.1 can cause a very noticable slowdown. This also goes for 0.01 but it is 10 times faster and will be enough waiting time in 95% (estimated) of the cases.

Reply:
Yeah, 0.1 seems a bit excessive. Adjusted to 0.01.